packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilter.tsx

```diff
@@ -0,0 +1,173 @@
+    cloneDeep(nextQueryRequest) as QueryBundleRequest,
+    facet,
+  )
+
+  const columnModel = data?.columnModels
+    ? getCorrespondingColumnForFacet(facet, data.columnModels)
+    : undefined
+
+  const userIds =
+    columnModel?.columnType === ColumnTypeEnum.USERID ||
+    columnModel?.columnType === ColumnTypeEnum.USERID_LIST
+      ? facet.facetValues.map(facet => facet.value)
+      : []
+  const userGroupHeaders = useGetInfoFromIds<UserGroupHeader>({
+    ids: userIds,
+    type: 'USER_PROFILE',
+  })
+
+  const entityIds =
+    columnModel?.columnType === ColumnTypeEnum.ENTITYID ||
+    columnModel?.columnType === ColumnTypeEnum.ENTITYID_LIST
+      ? facet.facetValues.map(facet => facet.value)
+      : []
+  const entityHeaders = useGetInfoFromIds<EntityHeader>({
+    ids: entityIds,
+    type: 'ENTITY_HEADER',
+  })
+
+  const evaluationIds =
+    columnModel?.columnType === ColumnTypeEnum.EVALUATIONID
+      ? facet.facetValues.map(facet => facet.value)
+      : []
+  const evaluations = useGetInfoFromIds<Evaluation>({
+    ids: evaluationIds,
+    type: 'EVALUATION_QUEUE',
+  })
+
+  const displayedFacetValues: RenderedFacetValue[] = useMemo(() => {
+    return sortBy(
+      facet.facetValues.map(
+        (facetValue: FacetColumnResultValueCount): RenderedFacetValue => {
+          return {
+            ...facetValue,
+            // Selected status should be based on the 'nextQuery', not the result data
+            // This ensures the checkboxes respond instantly to user interaction, like while waiting for multiple changes to debounce
+            isSelected:
+              currentSelectedFacet?.facetValues.includes(facetValue.value) ??
+              false,
+            displayText: valueToLabel(
+              facetValue,
+              userGroupHeaders,
+              entityHeaders,
+              evaluations,
+            ),
+          }
+        },
+      ),
+      fv => -1 * fv.count,
+    )
+  }, [
+    currentSelectedFacet?.facetValues,
+    entityHeaders,
+    evaluations,
+    facet.facetValues,
+    userGroupHeaders,
+  ])
+
+  if (!columnModel) {
+    return <></>
+  }
+
+  return (
+    <EnumFacetFilterUI
+      facetTitle={getColumnDisplayName(facet.columnName, facet.jsonPath)}
+      allIsSelected={allIsSelected}
+      facetValues={displayedFacetValues}
+      containerAs={containerAs}
+      dropdownType={dropdownType}
+      hideCollapsible={hideCollapsible}
+      onAddValueToSelection={value => {
+        addValueToSelectedFacet(
+          pick(facet, ['columnName', 'jsonPath']),
+          value,
+          { debounce: true },
+        )
+      }}
+      onRemoveValueFromSelection={value => {
+        removeValueFromSelectedFacet(
+          pick(facet, ['columnName', 'jsonPath']),
+          value,
+          { debounce: true },
+        )
+      }}
+      onRemoveAllFacetSelections={() =>
+        removeSelectedFacet(pick(facet, ['columnName', 'jsonPath']))
+      }
```

Comment:
Also transform these component-specific callbacks to the actions provided by QueryWrapper